apps/remote_control/lib/lexical/remote_control.ex

```diff
@@ -7,36 +7,29 @@ defmodule Lexical.RemoteControl do
 
   alias Lexical.Project
   alias Lexical.RemoteControl
+  alias Lexical.RemoteControl.ProjectNode
+  require Logger
 
   @allowed_apps ~w(common path_glob remote_control elixir_sense)a
 
   @app_globs Enum.map(@allowed_apps, fn app_name -> "/**/#{app_name}*/ebin" end)
 
-  @localhost_charlist '127.0.0.1'
-
   def start_link(%Project{} = project, project_listener) do
     :ok = ensure_epmd_started()
     entropy = :rand.uniform(65_536)
-
     start_net_kernel(project, entropy)
 
-    node_name = String.to_charlist("#{Project.name(project)}")
-    {:ok, paths} = system_paths(project)
-
-    erl_args =
-      erl_args([
-        "-hosts #{@localhost_charlist}",
-        "-setcookie #{Node.get_cookie()}",
-        "-sbwt none",
-        "-path #{paths}",
-        "-noshell"
-      ])
-
     apps_to_start = [:elixir | @allowed_apps] ++ [:runtime_tools]
     remote_control_config = Application.get_all_env(:remote_control)
 
-    with {:ok, node} <- :slave.start_link(@localhost_charlist, node_name, erl_args),
-         :ok <- :rpc.call(node, :code, :add_paths, [glob_paths()]),
+    node = node_name(project)
+    paths = format_prepending_paths(glob_paths())
+    start_options = [paths: paths, name: Atom.to_string(node), cookie: Node.get_cookie()]
```

Comment:
you don't need to convert name into a string; atoms convert without the leading colon.